jx/bdd/tf-boot/ci.sh

```diff
@@ -117,15 +117,22 @@ function authenticate() {
 function cluster_destroy()
 {
   echo "###############################################################################"
-	echo "Cleanup..."
-	pushd "$script_dir"/terraform
-	terraform destroy -auto-approve
-#	gcloud container clusters delete "$(yq r jx-requirements.yml 'cluster.clusterName')" --zone=europe-west1-c --quiet
-#	gsutil rm -r -f "$(yq r jx-requirements.yml 'storage.logs.url')"
-#	gsutil rm -r -f "$(yq r jx-requirements.yml 'storage.reports.url')"
-#	gsutil rm -r -f "$(yq r jx-requirements.yml 'storage.repository.url')"
-#	gsutil rm -r -f gs://"$(yq r jx-requirements.yml 'vault.bucket')"
-	popd
+  echo "Cleanup..."
+  pushd "$script_dir"/terraform
+
+  # stop at least the Vault instance to prevent Bank-Vault operator to interfere with deletion
+  kubectl delete vault "$(yq r jx-requirements.yml 'cluster.clusterName')"
+
+  # explicitly empty buckets (see https://github.com/jenkins-x/jx/issues/7406)
+  # see also https://github.com/GoogleCloudPlatform/gsutil/issues/417
+  gsutil -m -q rm -r -f "$(yq r jx-requirements.yml 'storage.logs.url')"/** || true
+  gsutil -m -q rm -r -f "$(yq r jx-requirements.yml 'storage.reports.url')"/**  || true
+  gsutil -m -q rm -r -f "$(yq r jx-requirements.yml 'storage.repository.url')"/**  || true
+  gsutil -m -q rm -r -f gs://"$(yq r jx-requirements.yml 'vault.bucket')"/**  || true
```

Comment:
That seems to do the trick regarding the vault bucket emptying. I think the `gsutil` is not even necessary, but I'll leave it for good measure.

Reply:
@abayer - https://github.com/jenkins-x/jx/pull/7411/files#diff-0d74d27cef898d7b54f6c3c824b267a2R256-R262

This might be another reason why we see resources not getting cleaned up. We are not destroying the cluster in case a test fails. In this case, it is up to the gc. I am wondering whether the current gc jobs properly capture and delete what Terraform creates. 

I had this morning three cases where the actual BDD tests failed, so I guess that left some stuff around.

Reply:
Yeah, that's a separate problem with things not getting cleaned up.